core/msd/src/test/java/com/yahoo/athenz/msd/CompositeInstanceTest.java

```diff
@@ -0,0 +1,156 @@
+
+        assertNotNull(instance);
+        assertEquals(instance.getDomainName(), "athenz");
+        assertEquals(instance.getServiceName(), "api");
+        assertEquals(instance.getInstance(), "i-1333w3er3rr");
+        assertEquals(instance.getInstanceType(), "ec2");
+        assertEquals(instance.getProvider(), "aws");
+        assertEquals(instance.getCertExpiryTime(), Timestamp.fromMillis(1722347760000L));
+        assertEquals(instance.getCertIssueTime(), Timestamp.fromMillis(1722260648000L));
+        assertEquals(instance, instance);
+
+        CompositeInstance instance2 = new CompositeInstance();
+        instance2.setDomainName("athenz")
+            .setServiceName("api")
+            .setInstance("i-1333w3er3rr")
+            .setInstanceType("ec2")
+            .setProvider("aws")
+            .setCertExpiryTime(Timestamp.fromMillis(1722347760000L))
+            .setCertIssueTime(Timestamp.fromMillis(1722260648000L));
+
+        assertEquals(instance, instance2);
+
+        instance2.setDomainName("sports");
+        assertNotEquals(instance, instance2);
+        instance2.setDomainName(null);
+        assertNotEquals(instance, instance2);
+
+        instance2.setDomainName("athenz");
+        instance2.setServiceName("apiv2");
+        assertNotEquals(instance, instance2);
+        instance2.setServiceName(null);
+        assertNotEquals(instance, instance2);
+
+        instance2.setServiceName("api");
+        instance2.setInstance("i-2423w3er3rr");
+        assertNotEquals(instance, instance2);
+        instance2.setInstance(null);
+        assertNotEquals(instance, instance2);
+
+        instance2.setInstance("i-1333w3er3rr");
+        instance2.setInstanceType("vm");
+        assertNotEquals(instance, instance2);
+        instance2.setInstanceType(null);
+        assertNotEquals(instance, instance2);
+
+        instance2.setInstanceType("ec2");
+        instance2.setProvider("gcp");
+        assertNotEquals(instance, instance2);
+        instance2.setProvider(null);
+        assertNotEquals(instance, instance2);
+
+        instance2.setProvider("aws");
+        instance2.setCertExpiryTime(Timestamp.fromMillis(1722247760000L));
+        assertNotEquals(instance, instance2);
+        instance2.setCertExpiryTime(null);
+        assertNotEquals(instance, instance2);
+
+        instance2.setCertExpiryTime(Timestamp.fromMillis(1722347760000L));
+        instance2.setCertIssueTime(Timestamp.fromMillis(1722230648000L));
+        assertNotEquals(instance, instance2);
+        instance2.setCertIssueTime(null);
+        assertNotEquals(instance, instance2);
+
+        instance2.setCertIssueTime(Timestamp.fromMillis(1722260648000L));
+        assertEquals(instance, instance2);
+
+        assertNotEquals(instance, null);
+        // for code coverage
+        assertFalse(instance.equals("mystring"));
+        assertNotEquals(instance, "mystring");
+
+        assertEquals(instance, instance);
+
+    }
+//
```

Comment:
any reason why we have all this commented code?

Reply:
my bad, forgot to remove that.